ouroboros-network-framework/src/Ouroboros/Network/Server2.hs

```diff
@@ -180,7 +194,30 @@ run ServerArguments {
       case result of
         (AcceptFailure err, acceptNext) -> do
           traceWith tracer (TrAcceptError err)
-          acceptLoop localAddress acceptNext
+          -- Try the determine if the connection was aborted by the remote end
+          -- before we could process the accept, or if it was a resource
+          -- exaustion problem.
+          -- NB. This piece of code is fragile and depends on specific
+          -- strings/mappings in the network and base libraries.
+          case fromException err of
+             Just ioErr ->
+#if defined(mingw32_HOST_OS)
+                      -- On Windows the network packet classifies all errors
+                      -- as OtherError. This means that we're forced to match
+                      -- on the error string. The text string comes from
+                      -- the network package's winSockErr.c, and if it ever
+                      -- changes we must update our text string too.
+                      if ioeGetErrorString ioErr /=
+                          "Software caused connection abort (WSAECONNABORTED)"
+                         then throwIO ioErr
+                         else threadDelay 0.5 >>
+                             acceptLoop localAddress acceptNext
```

Comment:
Let's define `iseCONNABORTED` for `mingw32_HOST_OS`.  This way we won't have to use `CPP`.

Reply:
I'll define `iseCONNABORTED` for windows but it we still need CPP. It just means that the string comparison will get moved to that function.

Reply:
Yes, we will still need CPP, but it will be just in one place.